packages/logger-plugin/tests/logger.plugin.spec.ts

```diff
@@ -69,93 +104,193 @@ describe('NgxsLoggerPlugin', () => {
+    // Assert
+    const expectedCallStack = [
+      ['group', 'action @@INIT (started @ )'],
+      ['log', '%c prev state', 'color: #9E9E9E; font-weight: bold', initialState],
+      ['log', '%c next state', 'color: #4CAF50; font-weight: bold', initialState],
+      ['groupEnd']
+    ];
+    expect(logger.getCallStack()).toEqual(expectedCallStack);
+  });
 
-    store.dispatch(new UpdateBarAction());
-
-    const expectedCallStack = LoggerSpy.createCallStack([
-      ...formatActionCallStack({ action: InitState.type, prevState: stateModelDefaults }),
+  it('should log success action with colors', () => {
+    // Arrange
+    const { store, logger } = setup([TestState]);
+    logger.clear();
+    const initialState = store.selectSnapshot(state => state);
 
-      ...formatActionCallStack({
-        action: UpdateBarAction.type,
-        prevState: stateModelDefaults,
-        nextState: { bar: defaultBarValue }
-      })
-    ]);
+    // Act
+    store.dispatch(new UpdateBarAction());
 
-    expect(logger.callStack).toEqual(expectedCallStack);
+    // Assert
+    const newState = store.selectSnapshot(state => state);
+    const expectedCallStack = [
+      ['group', 'action UPDATE_BAR (started @ )'],
+      ['log', '%c prev state', 'color: #9E9E9E; font-weight: bold', initialState],
+      ['log', '%c next state', 'color: #4CAF50; font-weight: bold', newState],
+      ['groupEnd']
+    ];
+    expect(logger.getCallStack()).toEqual(expectedCallStack);
   });
 
   it('should log success action with payload', () => {
+    // Arrange
     const { store, logger } = setup([TestState]);
+    logger.clear();
     const payload = 'qux';
 
+    // Act
     store.dispatch(new UpdateBarAction(payload));
 
-    const expectedCallStack = LoggerSpy.createCallStack([
-      ...formatActionCallStack({ action: InitState.type, prevState: stateModelDefaults }),
+    // Assert
+    const expectedCallStack = [
+      ['group', 'action UPDATE_BAR (started @ )'],
+      ['log', '%c payload', 'color: #9E9E9E; font-weight: bold', { bar: 'qux' }],
+      ['log', '%c prev state', 'color: #9E9E9E; font-weight: bold', { test: { bar: '' } }],
+      ['log', '%c next state', 'color: #4CAF50; font-weight: bold', { test: { bar: 'qux' } }],
+      ['groupEnd']
+    ];
+    expect(logger.getCallStack()).toEqual(expectedCallStack);
+  });
 
-      ...formatActionCallStack({
-        action: UpdateBarAction.type,
-        prevState: stateModelDefaults,
-        nextState: { bar: payload },
-        payload: { bar: payload }
-      })
-    ]);
+  it('should log async success action', async () => {
+    // Arrange
+    const { store, logger } = setup([TestState]);
+    logger.clear();
+    const payload = 'qux';
 
-    expect(logger.callStack).toEqual(expectedCallStack);
+    // Act
+    const promise = store.dispatch(new AsyncAction(payload)).toPromise();
+    logger.log('Some other work');
+    await promise;
+
+    // Assert
+    const expectedCallStack = [
```

Comment:
I suggest also adding a test that will call three asynchronous actions and see how it will look in the logs

Reply:
Great idea. Especially if they call each other.